src/neo/Network/P2P/Payloads/Conditions/CalledByGroupCondition.cs

```diff
@@ -0,0 +1,55 @@
+// Copyright (C) 2015-2021 The Neo Project.
+// 
+// The neo is free software distributed under the MIT software license, 
+// see the accompanying file LICENSE in the main directory of the
+// project or http://www.opensource.org/licenses/mit-license.php 
+// for more details.
+// 
+// Redistribution and use in source and binary forms with or without
+// modifications are permitted.
+
+using Neo.Cryptography.ECC;
+using Neo.IO;
+using Neo.IO.Json;
+using Neo.SmartContract;
+using Neo.SmartContract.Native;
+using System.IO;
+using System.Linq;
+
+namespace Neo.Network.P2P.Payloads.Conditions
+{
+    public class CalledByGroupCondition : WitnessCondition
+    {
+        /// <summary>
+        /// The group to be checked.
+        /// </summary>
+        public ECPoint Group;
+
+        public override int Size => base.Size + Group.Size;
+        public override WitnessConditionType Type => WitnessConditionType.CalledByGroup;
+
+        protected override void DeserializeWithoutType(BinaryReader reader)
+        {
+            Group = reader.ReadSerializable<ECPoint>();
+        }
+
+        public override bool Match(ApplicationEngine engine)
+        {
+            engine.ValidateCallFlags(CallFlags.ReadStates);
+            ContractState contract = NativeContract.ContractManagement.GetContract(engine.Snapshot, engine.CallingScriptHash);
+            return contract is not null && contract.Manifest.Groups.Any(p => p.PubKey == Group);
```

Comment:
ECPoint doesn't have override operator `==`, use `Equals` to avoid unexpected compare result.